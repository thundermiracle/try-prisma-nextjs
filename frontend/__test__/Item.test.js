import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import ItemComponent from "../components/Item";

const fakeItem = {
  id: "123",
  title: "A cool city",
  price: 50000000,
  description: "This price is not negotiable",
  image: "city.jpg",
  largeImage: "largeCity.jpg",
};

// enzyme test
describe("by enzyme test", () => {
  it("renders and displays properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);

    expect(wrapper.find("PriceTag").children().text()).toBe("Â¥50,000,000");
    expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
    expect(wrapper.find("img").prop("src")).toBe(fakeItem.image);
  });

  it("renders out the buttons properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);

    expect(wrapper.find(".buttonList").children()).toHaveLength(3);
    expect(wrapper.find(".buttonList").find("Link").exists()).toBeTruthy();
  });
});

describe("by snapshot", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
