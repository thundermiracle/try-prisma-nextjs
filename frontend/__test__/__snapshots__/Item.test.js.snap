// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`by snapshot matches snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool city"
    src="city.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <a>
        A cool city
      </a>
    </Link>
  </Title>
  <PriceTag>
    Â¥50,000,000
  </PriceTag>
  <p>
    This price is not negotiable
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <button>
        <a>
          Edit
        </a>
      </button>
    </Link>
    <AddToCart
      id="123"
    />
    <DeleteItem
      id="123"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
